docs/new/develop/indexes.md

```diff
@@ -0,0 +1,99 @@
+When you execute your Substreams for the first time, you are reading the data stored in the flat files of the Substreams provider. 
+
+To improve the performance, the data accessed by the Substreams is cached, so that the second or the third time that you run the Substreams, you can read from the cache, thus saving time and money. This behavior is illustrated in the following diagram.
+
+<figure><img src="../../.gitbook/assets/develop/cache-substreams-flow.png" width="100%" /></figure>
+
+This data indexing is done implicitly every time you run a Substreams for the first time, but Substreams also allows you to explicitly specify which specific data you want to index for further performance improvement.
+
+## Indexes
+
+Substreams has recently introduced the concept of _indexed modules_. An indexed module is a module that has been pre-cached for some specific data. Let's see it with an example!
+
+Consider that you want to retrieve all the Ethereum events matching a specific address. Usually, in every block, you would iterate through all the logs  looking for those where `log.address == ADDRESS`.
+
+With indexing, you could have a pre-cached module with the information of all the event addresses in the block. Instead of reading the full Ethereum block, you can search in the events index (event cache) and avoid decoding the data of those blocks that do not contain events you are interested in.
+
+In the following diagram, you can see three blocks with their corresponding data. Consider that you want to retrieve all the events where `log.address == 0xcd2...`. In a non-indexed module, you would have to decode the data of every block, iterate over the logs, and search if there is any log matching the condition.
+
+<figure><img src="../../.gitbook/assets/develop/cache-block-1.png" width="100%" /></figure>
+
+On the other hand, in an indexed module, the events of every block are pre-cached in a special store, so when you look for events where `log.address == 0xcd2...`, you can simply search in the index store of the block. If the event is contained, within the block, then you decode the data. If not, you skip it.
```

Comment:
in the following few paragraphs, there is confusion on two specific concepts:
1) the precaching of the "events" module
2) the index module that indicates where each event is contained...
I don't think it's ok as it is.


"In a non-indexed module, you would have to..." should be -> "without an index, you would have to go through the data of every block, but with an index, you can skip the blocks that do not contain the event that you want"